src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends Record<string, any>>(
```

Comment:
```suggestion
export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T>
```
do not use `any`, you need to specify the type